Strip TAB & SPACE from name & value passed to CookieStore methods

annevk opened the pull request:
#### 0b04262c9b8a699f5b21a185512f756e2b915a14
<pre>
Strip TAB &amp; SPACE from name &amp; value passed to CookieStore methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=297118">https://bugs.webkit.org/show_bug.cgi?id=297118</a>

Reviewed by Rupin Mittal.

The specification already covers set() and
<a href="https://github.com/whatwg/cookiestore/pull/281">https://github.com/whatwg/cookiestore/pull/281</a> will cover get(),
getAll(), and delete() for consistency.

subscribe() and unsubscribe() are not covered by this PR. They are not
shipping and have various outstanding specification issues.

This also removes an unused argument from
CookieStore::MainThreadBridge::set.

While using trim() I also discovered that the comments in
Source/WTF/wtf/text/StringImpl.cpp were out-of-date so this updates
those as well.

Canonical link: <a href="https://commits.webkit.org/298470@main">https://commits.webkit.org/298470@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da64dee537b10137c46536e94a0425eb8ec8a73a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115417 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35112 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25613 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121506 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65994 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3a725977-8683-4f00-8c54-cae8afabba5d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117306 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35770 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43682 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87695 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42375 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/8370e8b8-908b-4a29-aa98-a59eebcdf66e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118365 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28524 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103600 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68087 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/replace-get-all-cookies (failure)") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27682 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21722 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65159 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/107622 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97913 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21835 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124671 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/113956 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42356 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31732 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96473 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42723 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99788 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96259 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41490 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19346 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38260 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18479 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42236 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47792 "Built successfully") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/139053 "Build was cancelled. Recent messages:Failed to print configuration") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41738 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/139053 "Build was cancelled. Recent messages:Failed to print configuration") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45066 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43457 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/da64dee537b10137c46536e94a0425eb8ec8a73a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49117

RupinMittal commented on Source/WebCore/Modules/cookie-store/CookieStore.cpp:
```diff
@@ -234,6 +234,13 @@ CookieStore::~CookieStore()
+static String normalize(const String& string)
+{
+    if (string.contains(isTabOrSpace<char16_t>))
+        return string.trim(isTabOrSpace);
```
Nit: Perhaps we should specify `<char16_t>` here too for the sake of clarity and consistency with the above line? Not a big deal. Up to you.

RupinMittal replied:
Fair enough. Is it required in `contains`?

annevk replied:
Yeah, unfortunately. Not entirely sure why.

RupinMittal reviewed (approved):
LGTM

annevk committed:
Strip TAB & SPACE from name & value passed to CookieStore methods
https://bugs.webkit.org/show_bug.cgi?id=297118

Reviewed by Rupin Mittal.

The specification already covers set() and
https://github.com/whatwg/cookiestore/pull/281 will cover get(),
getAll(), and delete() for consistency.

subscribe() and unsubscribe() are not covered by this PR. They are not
shipping and have various outstanding specification issues.

This also removes an unused argument from
CookieStore::MainThreadBridge::set.

While using trim() I also discovered that the comments in
Source/WTF/wtf/text/StringImpl.cpp were out-of-date so this updates
those as well.

Canonical link: https://commits.webkit.org/298470@main
